src/dns.h

```diff
@@ -617,125 +682,94 @@ hsk_dns_name_read_size(
   const hsk_dns_dmp_t *dmp
 );
 
+/**
+ * Verify a single label (TLD) follows Handshake protocol rules
+ */
 bool
-hsk_dns_name_alloc(
-  uint8_t **data,
-  size_t *data_len,
-  const hsk_dns_dmp_t *dmp,
-  char **name
-);
-
-bool
-hsk_dns_name_dirty(const char *name);
-
-void
-hsk_dns_name_sanitize(const char *name, char *out);
+hsk_dns_name_verify(const uint8_t *name);
 
-bool
-hsk_dns_name_verify(const char *name);
+int
+hsk_dns_name_cmp(const uint8_t *a, const uint8_t *b);
 
 bool
-hsk_dns_name_is_fqdn(const char *name);
-
-int
-hsk_dns_name_cmp(const char *a, const char *b);
+hsk_dns_name_equal(const void *a, const void *b);
 
 /**
  * Returns: number of labels found in the name
- * In:      name:   pointer to string containing full domain name
-            size:   max number of labels expected in the name
+ * In:      name:   pointer to byte array containing full domain name
+ *          size:   max number of labels expected in the name
  * Out:     labels: if present, array is filled with offset (in bytes)
  *                  of starting point of each label in name
  */
 int
-hsk_dns_label_split(const char *name, uint8_t *labels, size_t size);
+hsk_dns_label_split(const uint8_t *name, uint8_t *labels, size_t size);
 
 /**
  * Returns: number of labels in the name
- * In:      name:   pointer to string containing full domain name
+ * In:      name:   pointer to byte array containing full domain name
  */
 int
-hsk_dns_label_count(const char *name);
+hsk_dns_label_count(const uint8_t *name);
 
 /**
- * Returns: length of string from requested label through end of name
- * In:      name:   pointer to string containing full domain name
+ * Returns: length of byte array from requested label through end of name
+ *          this includes all length bytes and terminal 0x00 byte.
+ * In:      name:   pointer to byte array containing full domain name
  *          labels: pointer to array of offsets of each label in name
  *          count:  number of labels in the name
  *          index:  index of starting label in name to return
  *                  (-1 = last label, TLD)
- * Out:     ret:    pointer to string to be filled with name from
- *                  specified label through end of name
+ * Out:     ret:    pointer to byte array to be filled with name from
+ *                  specified label through end of name including final 0x00
  */
 int
 hsk_dns_label_from2(
```

Comment:
I think name of this method is misleading, it returns sub name - not a specific label.

Reply:
Yeah it returns a name "starting from a given label" i.e. "label from". I suppose we could rename it `hsk_dns_name_from_label()` ?